File: c/912_Sort_an_Array.py
from typing import List

class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        return self.mergeSort(nums,0,len(nums)-1)

    def mergeLists(self, arr1,arr2):
        mergeArr=[]
        i,j=0,0
        n1,n2=len(arr1),len(arr2)

        while i<n1 and j<n2:
            if arr1[i] <= arr2[j]:
                mergeArr.append(arr1[i]);i+=1
            else:
                mergeArr.append(arr2[j]);j+=1
        
        while i<n1:
            mergeArr.append(arr1[i]);i+=1
        while j<n2:
            mergeArr.append(arr2[j]);j+=1

        return mergeArr

    def mergeSort(self,nums,left,right):
        if left<right:
            mid=(left+right)//2
            
            arr1=self.mergeSort(nums,left,mid)
            arr2=self.mergeSort(nums,mid+1,right)
            
            return self.mergeLists(arr1,arr2)
        
        return nums[left:right+1]

s=Solution()
print(s.sortArray([1,2,4,2,1]))

File: c/912_Sort_an_Array.js
const merge =
    (left, right) => {
      let res = [], l = 0, r = 0, i = 0;

      while (l < left.length && r < right.length) {
        if (left[l] <= right[r])
          res[i++] = left[l++];
        else
          res[i++] = right[r++];
      }

      return res.concat(left.slice(l), right.slice(r));
    }

const mergeSort = (nums, l, r) => {
  if (l === r - 1) return [nums[l]];

  let mid = (l + r) / 2 | 0;

  let left = mergeSort(nums, l, mid);
  let right = mergeSort(nums, mid, r);

  return merge(left, right);
};

var sortArray = function(nums) {
  return mergeSort(nums, 0, nums.length);
};

let res = sortArray([1, 5, 3, 4, 2]);

console.log(res);